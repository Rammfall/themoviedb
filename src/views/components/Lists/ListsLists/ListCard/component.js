import React from 'react'
import PropTypes from 'prop-types'
import { Card, Col, Typography } from 'antd'
import { DeleteOutlined } from '@ant-design/icons'

const ListCard = ({ name, description }) => (
  <Col
    xs={{ span: 24 }}
    sm={{ span: 12 }}
    md={{ span: 8 }}
    lg={{ span: 8 }}
    xl={{ span: 6 }}
  >
    <Card
      hoverable
      className="top-margin"
      actions={[
        <DeleteOutlined
          key="delete"
        />
      ]}
    >
      <Typography.Title level={4}>
        {name}
      </Typography.Title>
      {description}
    </Card>
  </Col>
)

ListCard.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
}

export default ListCard
