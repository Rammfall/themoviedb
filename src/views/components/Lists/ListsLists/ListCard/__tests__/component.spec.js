import React from 'react'
import { shallow } from 'enzyme'

import ListCard from '../component'

describe('ListCard', () => {
  it('matches snapshot', () => {
    const props = {
      name: 'test',
      description: 'test',
      onDelete: jest.fn(),
      id: '3'
    }
    const component = shallow(<ListCard {...props} />)

    expect(component).toMatchSnapshot()
  })
})
