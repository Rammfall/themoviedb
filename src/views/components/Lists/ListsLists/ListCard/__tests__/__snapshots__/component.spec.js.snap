// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard matches snapshot 1`] = `
<Col
  lg={
    Object {
      "span": 8,
    }
  }
  md={
    Object {
      "span": 8,
    }
  }
  sm={
    Object {
      "span": 12,
    }
  }
  xl={
    Object {
      "span": 6,
    }
  }
  xs={
    Object {
      "span": 24,
    }
  }
>
  <Card
    actions={
      Array [
        <ForwardRef(DeleteOutlined)
          onClick={[Function]}
        />,
      ]
    }
    className="top-margin"
    hoverable={true}
  >
    <Title
      level={4}
    >
      test
    </Title>
    test
  </Card>
</Col>
`;
